//importamos modelo
import getTrainingUsersModel from '../../models/trainings/getTrainingUsersModel.js';

//importamos la funci贸n que lanza errores
import generateErrorUtil from '../../utils/generateErrorUtil.js';

//funci贸n que obtiene los usuarios de un entrenamiento
const getTrainingUsersController = async (req, res, next) => {
    try {
        //obtenemos el id del entrenamiento
        const { trainingId } = req.params;

        //verificamos que se proporcion贸 el id del entrenamiento
        if (!trainingId) {
            generateErrorUtil('Falta el id del entrenamiento', 400);
        }

        //obtenemos los usuarios del entrenamiento
        const users = await getTrainingUsersModel(trainingId);

        //enviamos respuesta
        res.send({
            status: 'ok',
            data: users,
        });
    } catch (err) {
        next(err);
    }
};
//exportamos la funci贸n
export default getTrainingUsersController;
